modules/storages/app/components/storages/open_project_storage_modal_component.rb

```diff
@@ -29,15 +29,13 @@
 #++
 
 class Storages::OpenProjectStorageModalComponent < ViewComponent::Base
+  attr_reader :controller, :project_storage_open_url, :redirect_url, :state
+
   def initialize(project_storage_open_url:, redirect_url:, state:, **options)
     super
-    controller = "storages--open-project-storage-modal"
-    @data = {
-      controller:,
-      "application-target": "dynamic",
-      "#{controller}-project-storage-open-url-value": project_storage_open_url,
-      "#{controller}-redirect-url-value": redirect_url
-    }
```

Comment:
I found out that trying to call a `render` during initialization explodes everything. So I needed to refactor this guy a bit.

Functionality *should* be the same as before.

Reply:
nice find